### PR TITLE
Convert string refs to callback refs

### PR DESCRIPTION
**Pull request by athomann**
**Summary**
Convert string refs to callback refs based on React team statements of likely future deprecation of string refs.  https://facebook.github.io/react/docs/more-about-refs.html#the-ref-string-attribute

**Test Plan**
No additional testing needed.

### CODE REVIEW

**Comment by facebook-github-bot-5** [comment]
Thank you for your pull request and welcome to our community. We require contributors to sign our Contributor License Agreement, and we don't seem to have you on file. In order for us to review and merge your code, please sign up at https://code.facebook.com/cla - and if you have received this in error or have any questions, please drop us a line at cla@fb.com. Thanks!

**Comment by facebook-github-bot-0** [comment]
Thank you for signing our Contributor License Agreement. We can now accept your code for this (and any) Facebook open source project. Thanks!

File: src/component/base/DraftEditor.react.js [hunk 1]
```diff
@@ -110,6 +110,8 @@ class DraftEditor extends React.Component {
   _onPaste: Function;
   _onSelect: Function;
 
+  editor: () => void;
+  editorContainer: () => void;
```
Comment (hellendag) [inline src/component/base/DraftEditor.react.js]: These should be React components, not functions.

File: src/component/contents/DraftEditorLeaf.react.js [hunk 2]
```diff
@@ -73,6 +73,9 @@ class DraftEditorLeaf extends React.Component {
    * DOM structure of the TextEditor component. If leaves had multiple
    * text nodes, this would be harder.
    */
+
+  leaf: () => void;
```
Comment (hellendag) [inline src/component/contents/DraftEditorLeaf.react.js]: This is a span.

**Comment by hellendag** [comment]
Thanks for doing this! Just a comment about the Flow types, which I suppose aren't currently being checked properly...

File: examples/tex/js/components/TeXBlock.js [hunk 3]
```diff
@@ -32,7 +32,7 @@ class KatexOutput extends React.Component {
     this._timer = setTimeout(() => {
       katex.render(
         this.props.content,
-        this.refs.container,
+        this.container,
```
Comment (zpao) [inline examples/tex/js/components/TeXBlock.js]: You'll need to update the actual ref in render too to be a callback. Looks like you just missed this one.

**Comment by athomann** [comment]
@hellendag Thanks, I'll make adjustments.  I'm new flowtype, I would change () => void to editor: React.Element<any> ?  and leaf: span ?

**Comment by athomann** [comment]
Added revision.

**Comment by danielr18** [comment]
Will this be merged for v0.11 ?

**Comment by flarnie** [comment]
Wow! Glad I just came across this - super happy to have this change! Looks like it just needs rebased. I can try to tidy this up, and then we'd be happy to merge it. :)

**Comment by athomann** [comment]
Hey @flarnie, I rebased this PR.  LMK if there is anything else that needs to change.

**Comment by athomann** [comment]
I've also added you as a contributor to my branch if you'd like to change anything and I'm not available.

**Comment by flarnie** [comment]
Thanks! Seems like we just need to resolve the flow errors. I will look at that when I get a chance, either tonight or tomorrow morning, unless you beat me to it. :)

**Comment by athomann** [comment]
@flarnie Is it preferred to use a maybe type `?HTMLElement` or only assign the ref if not null?

**Comment by athomann** [comment]
Other parts of the src use maybe types, so I opted for that.

**Comment by flarnie** [comment]
Interesting - this change seems to make one of our internal integration tests fail. I will look into it more - still want to get this in but need to resolve the test failure before landing.

**Comment by flarnie** [comment]
Manually testing and re-running the tests locally, things look ok to me. I don't expect anyone to be depending on these refs outside of Draft, hopefully I'm right. Will try landing this again on Monday. 🍀

**Comment by athomann** [comment]
Thanks for the update!  LMK if there is anything I can do.

**Pull request closed without merging**